/*
 * This file was automatically generated by EvoSuite
 * Fri Sep 08 16:31:15 GMT 2017
 */

package com.araguacaima.commons.exception.core;

import org.junit.Test;
import static org.junit.Assert.*;
import com.araguacaima.commons.exception.core.ApplicationException;
import com.araguacaima.commons.exception.core.Severity;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class ApplicationException_ESTest extends ApplicationException_ESTest_scaffolding {

  @Test
  public void test0()  throws Throwable  {
      Severity severity0 = Severity.ERROR;
      ApplicationException applicationException0 = new ApplicationException("", severity0, "", (Object) null);
      assertEquals("", applicationException0.getKeyCode());
  }

  @Test
  public void test1()  throws Throwable  {
      Severity severity0 = Severity.ERROR;
      Object[] objectArray0 = new Object[4];
      ApplicationException applicationException0 = new ApplicationException("", severity0, objectArray0);
      assertEquals("", applicationException0.getKeyCode());
  }

  @Test
  public void test2()  throws Throwable  {
      ApplicationException applicationException0 = new ApplicationException("");
      assertEquals("", applicationException0.getExtendedMessage());
  }

  @Test
  public void test3()  throws Throwable  {
      Severity severity0 = Severity.ERROR;
      ApplicationException applicationException0 = new ApplicationException("", severity0);
      ApplicationException applicationException1 = new ApplicationException(" ", applicationException0);
      assertFalse(applicationException1.equals((Object)applicationException0));
  }

  @Test
  public void test4()  throws Throwable  {
      Severity severity0 = Severity.ERROR;
      ApplicationException applicationException0 = new ApplicationException((String) null, severity0, "");
      assertEquals("", applicationException0.getExtendedMessage());
  }

  @Test
  public void test5()  throws Throwable  {
      Severity severity0 = Severity.ERROR;
      ApplicationException applicationException0 = new ApplicationException("", severity0);
      ApplicationException applicationException1 = new ApplicationException("", severity0, applicationException0);
      assertFalse(applicationException1.equals((Object)applicationException0));
  }
}
